src/search.rs

```diff
@@ -198,18 +194,81 @@ impl Search {
     Ok(directory.to_owned())
   }
 
-  fn working_directory_from_justfile(justfile: &Path) -> SearchResult<PathBuf> {
-    Ok(
-      justfile
-        .parent()
-        .ok_or_else(|| SearchError::JustfileHadNoParent {
-          path: justfile.to_path_buf(),
-        })?
-        .to_owned(),
-    )
+  /// Returns the path to the directory where the jusfile is located.
+  /// Returns an empty path if the justfile path is relative and has only one component.
+  /// Returns an empty path if the justfile path is itself empty.
+  /// Returns the current working directory if the justfile is not a file on disk.
+  pub fn justfile_directory(&self) -> PathBuf {
```

Comment:
These functions are kind of footguns. If the justfile was read from stdin, then there is no path or directory to the justfile. I understand that these are kind of compatibility shims, but I think the code just has to be adapted to deal with the fact that the justfile path might be None.

Reply:
My reasoning is that you still need to know the working directory even if you load the justfile from stdin. On the other hand, many functions don't need to know both the working directory and the optional justfile path. That's what this function does. It abstracts away the justfile directory over the `Search` struct. I don't have strong opinion on this, but the abstraction seems reasonable. How do you see it?

Reply:
`working_directory` is fine, and should always be set. It's `justfile` and `justfile_directory` that I think are the issue, since there is no justfile, and no directory containing it.